components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java

```diff
@@ -139,8 +142,13 @@ protected void doStop() throws Exception {
                 LOG.debug("Shutting down Kafka consumer worker threads with timeout {} millis", timeout);
                 getEndpoint().getCamelContext().getExecutorServiceManager().shutdownGraceful(executor, timeout);
             } else {
-                executor.shutdownNow();
+                executor.shutdown();
             }
+
+            if (!executor.awaitTermination(5, TimeUnit.SECONDS)) {
+                LOG.warn("The tasks did not finish within the specified time");
```

Comment:
This WARN message needs more details so end users can understand it. Also the 5 sec is hardcoded